command/agent/auth/auth.go

```diff
@@ -364,10 +424,15 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 			return err
 		}
 
-		// Start the renewal process
-		ah.logger.Info("starting renewal process")
 		metrics.IncrCounter([]string{"agent", "auth", "success"}, 1)
-		go watcher.Renew()
+		// We don't want to trigger the renewal process for tokens with
+		// unlimited TTL, such as the root token.
+		if leaseDuration == 0 && isTokenFileMethod {
```

Comment:
Don't we have an actual renewable field we can use to make this determination instead?

Reply:
We do, for non-renewable tokens we still seem to rely on some behaviour in lifetimewatcher:
```
		case nonRenewable || r.renewBehavior == RenewBehaviorRenewDisabled:
			// Can't or won't renew, just keep the same expiration so we exit
			// when it's reauthentication time
			remainingLeaseDuration = fallbackLeaseDuration
```

In particular, we care about whether it's non-renewable _and_ the leaseDuration is 0, which I guess might be a better test than the existing one

Reply:
Hmm, making this change, it seems to break tests including tokens with wrapping, though it could be that I don't have enough context about token wrapping as of right now. I'll dig into this a little more I think, but the current behaviour doesn't break anything, at least

Reply:
Okay, right, this approach falls apart a little with wrapping tokens, as it doesn't seem that you can use them to authenticate. i.e. you get a 403 `bad token` response from Vault, and ultimately, adding the results of a wrapped `auth/token/create` to your token file would not work, today, with this code. 

I might be missing something, but while I think it'd probably be possible to add a new config value that unwraps the token for you, I can't see another elegant way to do this kind of user flow. Likewise, whether or not we want to do that is another question.

Reply:
I think that if this behaviour seems acceptable (you have more context than me with regards to wrapping tokens!) and we think this PR is in a mergeable state, I think it makes sense to merge like it is, and I can add a note about this in the docs. Do let me know if there's anything else you'd like to see before I merge this, though!